src/praisonai-agents/praisonaiagents/tools/messaging_tools.py

```
@@ -139,6 +139,54 @@ def _parse_media(message: str) -> tuple[str, Optional[List[str]]]:
 
+def _check_send_policy(target: str) -> Optional[str]:
+    """Evaluate the active send-policy for ``target`` (Issue #2226).
+
+    Returns a model-readable denial string when the send is *not* permitted, or
+    ``None`` when allowed (no policy registered, or policy allows). The policy
+    is consulted from the task-local context and is enriched with the current
+    session's agent/session/origin so back-ends can scope decisions.
+    """
+    try:
+        from ..session.context import get_send_policy, get_session_context
+    except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)